### PR TITLE
python311Packages.pygls: 1.0.2 -> 1.1.1

### PR DESCRIPTION
**Pull request by kalekseev**
#### pygls ✅
Diff https://github.com/openlawlibrary/pygls/compare/v1.0.2...v1.1.1

Main change: switched to poetry build.
Also they depend on typeguard v3 we have v4 but breaking changes in typeguard shouldn't affect pygls.

closes #233862

#### cmake-language-server ✅

diff https://github.com/regen100/cmake-language-server/compare/v0.1.7...v0.1.8

Switched to pdb-backend

#### jedi-language-server
Merged pygls 1.1 but haven't released it yet https://github.com/pappasam/jedi-language-server/commit/c4c470cff67e54593a626b22d1b6b05e56fde3a3

#### ruff-lsp ✅

diff https://github.com/astral-sh/ruff-lsp/compare/v0.0.40...v0.0.42

closes https://github.com/NixOS/nixpkgs/pull/261953

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 262733` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>cmake-language-server</li>
    <li>cmake-language-server.dist</li>
    <li>python310Packages.jedi-language-server</li>
    <li>python310Packages.jedi-language-server.dist</li>
    <li>python311Packages.jedi-language-server</li>
    <li>python311Packages.jedi-language-server.dist</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vscode-extensions.ms-python.python</li>
    <li>ycmd</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>bitbake-language-server</li>
    <li>bitbake-language-server.dist</li>
    <li>python310Packages.pygls</li>
    <li>python310Packages.pygls.dist</li>
    <li>python311Packages.pygls</li>
    <li>python311Packages.pygls.dist</li>
    <li>ruff-lsp</li>
    <li>ruff-lsp.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/jedi-language-server/default.nix [hunk 1]
```diff
@@ -16,16 +16,16 @@
 
 buildPythonPackage rec {
   pname = "jedi-language-server";
-  version = "0.41.0";
+  version = "0.41.2-dev";
```
Comment (kalekseev) [inline pkgs/development/python-modules/jedi-language-server/default.nix:19]: can we use unreleased version?

**Comment by kalekseev** [comment]
Result of `nixpkgs-review pr 262733` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>bitbake-language-server</li>
    <li>bitbake-language-server.dist</li>
    <li>cmake-language-server</li>
    <li>cmake-language-server.dist</li>
    <li>python310Packages.jedi-language-server</li>
    <li>python310Packages.jedi-language-server.dist</li>
    <li>python310Packages.pygls</li>
    <li>python310Packages.pygls.dist</li>
    <li>python311Packages.jedi-language-server</li>
    <li>python311Packages.jedi-language-server.dist</li>
    <li>python311Packages.pygls</li>
    <li>python311Packages.pygls.dist</li>
    <li>ruff-lsp</li>
    <li>ruff-lsp.dist</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vscode-extensions.ms-python.python</li>
    <li>ycmd</li>
  </ul>
</details>

File: pkgs/development/python-modules/jedi-language-server/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/python-modules/jedi-language-server/default.nix:19]: Mark this as `0.41.x-unstable-xxxx-xx-xx`.

**Review by doronbehar — APPROVED** [review]
Great work! Diff LGTM, and CI is green. Merging.

**Pull request merged**